pkg/types/position.go

```diff
@@ -218,14 +218,22 @@ func (p *Position) GetBase() (base fixedpoint.Value) {
 	return base
 }
 
-// GetBaseAndAverageCost locks the mutex and return the base quantity and average cost
-func (p *Position) GetBaseAndAverageCost() (base fixedpoint.Value, averageCost fixedpoint.Value) {
+type PositionStatus struct {
+	Base        fixedpoint.Value
+	AverageCost fixedpoint.Value
+	Type        PositionType
+}
+
+// GetStatus locks the mutex and return the position status
+func (p *Position) GetStatus() PositionStatus {
```

Comment:
why?

Reply:
I think the issue lies in the need for a thread-safe way to access position-related information,
which is used to determine trading conditions.

Perhaps, just like we discussed offline,
renaming it to "snapshot" would make it easier to understand.

Reply:
I leave the core API unchanged and move the locking util function to a private module in `xgap` folder instead.